osu.Game.Rulesets.Taiko/UI/TaikoPlayfieldAdjustmentContainer.cs

```diff
@@ -1,30 +1,72 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
-
 using System;
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Taiko.UI
 {
+    /// <summary>
+    /// Methods of which <see cref="TaikoPlayfieldAdjustmentContainer"/> adjusts the aspect ratio visually. This does not
+    /// affect time range directly.
+    /// </summary>
+    public enum AspectRatioAdjustmentMethod
+    {
+        /// <summary>
+        /// Do not adjust the aspect ratio of the playfield visually. Relative height will be kept at the default of
+        /// <see cref="TaikoPlayfieldAdjustmentContainer.default_relative_height"/>, and width will always expanded to
+        /// fill parent fully.
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// Adjust the playfield's size proportionally keeping the aspect ratio, This is the default lazer behaviour.
+        /// </summary>
+        Scale,
+
+        /// <summary>
+        /// Trim the playfield to fit the aspect ratio. If the display aspect ratio is narrower than the designated aspect
+        /// ratio, the playfield will extend beyond the display area, and thus be cropped.
+        /// </summary>
+        Trim
+    }
+
     public class TaikoPlayfieldAdjustmentContainer : PlayfieldAdjustmentContainer
     {
         private const float default_relative_height = TaikoPlayfield.DEFAULT_HEIGHT / 768;
-        private const float default_aspect = 16f / 9f;
+        public const float DEFAULT_ASPECT = 16f / 9f;
 
-        public readonly IBindable<bool> LockPlayfieldAspect = new BindableBool(true);
+        /// <summary>
+        /// The target aspect ratio for the playfield. This only affects display directly and not the time range, except
+        /// in the case of <see cref="AspectRatioAdjustmentMethod.None"/>, where the time range is scaled to have a
+        /// constant scroll rate.
```

Comment:
This xmldoc is breaking my brain for some reason. As far as I understand, the time range is never changed by this setting, `None` does nothing at all, `Scale` scales up the playfield to the target ratio and `Trim` hard-cuts the playfield at the target ratio. So what's this "except the `None` case" stuff? Is this an allusion to what `TaikoModClassic` is doing? If so then I'd just move that there.

Reply:
Yeah `None` doesn't do anything or change the time range in `TaikoPlayfieldAdjustmentContainer`, removed this part of the comment in abac44acc938fa90caa53a16ac0af5f172f3a8f7 and 08606a672668cb213c866fb8313234b3c2ca3eb4